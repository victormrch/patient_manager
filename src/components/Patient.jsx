import React from "react";

const Patient = () => {
  return (
    <div className="m-3 bg-white shadow-md px-5 py-10 rounded-xl">
      <p className="font-bold mb-3 text-gray-700 uppercase">
        Nombre : {""}
        <span className="font-normal normal-case">Hook</span>
      </p>
      <p className="font-bold mb-3 text-gray-700 uppercase">
        Owner Name : {""}
        <span className="font-normal normal-case">Stuart James</span>
      </p>
      <p className="font-bold mb-3 text-gray-700 uppercase">
        Email Address : {""}
        <span className="font-normal normal-case">stuart@jll.com</span>
      </p>
      <p className="font-bold mb-3 text-gray-700 uppercase">
        DISCHARGE DATE : {""}
        <span className="font-normal normal-case">21-01-2022</span>
      </p>
      <p className="font-bold mb-3 text-gray-700 uppercase">
        Reason for visit : {""}
        <span className="font-normal normal-case">
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Doloremque
          quibusdam cumque illo magnam, reprehenderit eligendi commodi ullam et
          soluta ratione odio. Minima neque aliquam, consequatur perferendis
          modi ipsa asperiores quas!
        </span>
      </p>
    </div>
  );
};

export default Patient;
