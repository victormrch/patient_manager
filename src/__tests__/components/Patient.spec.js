import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Patient from "../../components/Patient";
import "@testing-library/jest-dom/extend-expect";

const mockPatient = {
  dischargeDate: "2019-01-01",
  email: "test@test.com",
  id: "1234",
  name: "Test Name",
  owner: "Test Owner",
  reason: "Test Reason",
};

describe("Form component specs", () => {
  it("should render correctly using snapshot testing", () => {
    const { asFragment } = render(<Patient patient={mockPatient} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display the Patient Component", () => {
    render(<Patient patient={mockPatient} />);

    const elementDischargeDate = screen.getByText("2019-01-01");
    const elementEmail = screen.getByText("test@test.com");
    const elementName = screen.getByText("Test Name");
    const elementOwner = screen.getByText("Test Owner");
    const elementReason = screen.getByText("Test Reason");
    const elementEditButton = screen.getByRole("button", { name: "Edit" });
    const elementDeleteButton = screen.getByRole("button", { name: "Delete" });

    expect(elementDischargeDate).not.toBeNull();
    expect(elementDischargeDate).toBeInTheDocument();
    expect(elementDischargeDate.tagName).toEqual("SPAN");
    expect(elementDischargeDate.textContent).toEqual("2019-01-01");

    expect(elementEmail).not.toBeNull();
    expect(elementEmail).toBeInTheDocument();
    expect(elementEmail.tagName).toEqual("SPAN");
    expect(elementEmail.textContent).toEqual("test@test.com");

    expect(elementName).not.toBeNull();
    expect(elementName).toBeInTheDocument();
    expect(elementName.tagName).toEqual("SPAN");
    expect(elementName.textContent).toEqual("Test Name");

    expect(elementOwner).not.toBeNull();
    expect(elementOwner).toBeInTheDocument();
    expect(elementOwner.tagName).toEqual("SPAN");
    expect(elementOwner.textContent).toEqual("Test Owner");

    expect(elementReason).not.toBeNull();
    expect(elementReason).toBeInTheDocument();
    expect(elementReason.tagName).toEqual("SPAN");
    expect(elementReason.textContent).toEqual("Test Reason");

    expect(elementEditButton).not.toBeNull();
    expect(elementEditButton).toBeInTheDocument();
    expect(elementEditButton.tagName).toEqual("BUTTON");
    expect(elementEditButton.textContent).toEqual("Edit");

    expect(elementDeleteButton).not.toBeNull();
    expect(elementDeleteButton).toBeInTheDocument();
    expect(elementDeleteButton.tagName).toEqual("BUTTON");
    expect(elementDeleteButton.textContent).toEqual("Delete");
  });

  it("when EditButton is clicked, setPatient should be called one time per click", () => {
    const props = {
      patient: mockPatient,
      setPatient: jest.fn(),
      deletePatient: jest.fn(),
    };

    render(<Patient {...props} />);

    const elementEditButton = screen.getByRole("button", { name: "Edit" });

    expect(props.setPatient).toHaveBeenCalledTimes(0);

    userEvent.click(elementEditButton);

    expect(props.setPatient).toHaveBeenCalled();
    expect(props.setPatient).toHaveBeenCalledTimes(1);
  });

  it("when DeleteButton is clicked, deletePatient should be called one time per click and when confirm the delete", () => {
    const props = {
      patient: mockPatient,
      setPatient: jest.fn(),
      deletePatient: jest.fn(),
    };
    window.confirm = jest.fn(() => true);

    render(<Patient {...props} />);

    const elementDeleteButton = screen.getByRole("button", { name: "Delete" });

    expect(props.deletePatient).toHaveBeenCalledTimes(0);

    userEvent.click(elementDeleteButton);

    expect(props.deletePatient).toHaveBeenCalled();
    expect(props.deletePatient).toHaveBeenCalledTimes(1);
  });

  it("when DeleteButton is clicked, handleDelete should be called one time per click and shouldn't called deletePatient", () => {
    const props = {
      patient: mockPatient,
      setPatient: jest.fn(),
      deletePatient: jest.fn(),
    };
    window.confirm = jest.fn(() => false);

    const wrapper = render(<Patient {...props} />);

    const elementDeleteButton = screen.getByRole("button", { name: "Delete" });

    expect(props.deletePatient).toHaveBeenCalledTimes(0);

    userEvent.click(elementDeleteButton);

    expect(props.deletePatient).not.toHaveBeenCalled();
    expect(props.deletePatient).toHaveBeenCalledTimes(0);
  });
});
