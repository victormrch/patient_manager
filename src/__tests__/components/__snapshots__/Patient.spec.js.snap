// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component spects should render correctly using snapshot testing with Empty Patient 1`] = `
<DocumentFragment>
  <div
    class="my-5 mx-5 bg-white shadow-md px-5 py-10 rounded-xl"
  >
    <p
      class="font-bold mb-3 text-gray-700 uppercase"
    >
      Nombre : 
      <span
        class="font-normal normal-case"
      >
        Test Name
      </span>
    </p>
    <p
      class="font-bold mb-3 text-gray-700 uppercase"
    >
      Owner Name : 
      <span
        class="font-normal normal-case"
      >
        Test Owner
      </span>
    </p>
    <p
      class="font-bold mb-3 text-gray-700 uppercase"
    >
      Email Address : 
      <span
        class="font-normal normal-case"
      >
        test@test.com
      </span>
    </p>
    <p
      class="font-bold mb-3 text-gray-700 uppercase"
    >
      DISCHARGE DATE : 
      <span
        class="font-normal normal-case"
      >
        2019-01-01
      </span>
    </p>
    <p
      class="font-bold mb-3 text-gray-700 uppercase"
    >
      Reason for visit : 
      <span
        class="font-normal normal-case"
      >
        Test Reason
      </span>
    </p>
    <div
      class="flex flex-row justify-between mt-8"
    >
      <button
        class="py-2 px-10 mb-2 bg-amber-500 hover:bg-amber-700 text-white font-bold uppercase rounded-lg"
        type="button"
      >
        Edit
      </button>
      <button
        class="py-2 px-10 mb-2 bg-red-500 hover:bg-red-700 text-white font-bold uppercase rounded-lg"
        type="button"
      >
        Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
